### PR TITLE
chore: move env-builder to pkgdb

### PR DESCRIPTION
**Pull request by aakropotkin**
## Proposed Changes
- chore: `env-builder` code to pkgdb.
- feat: add `pkgdb buildenv` subcommand.
- chore: migrate static assets into `pkgdb`
- chore: remove `flox-env-builder` drv
- doc: updated `CONTRIBUTING.md` and `params-to-sql.md`

## Release Notes
N/A

### CODE REVIEW

**Comment by garbas** [comment]
🚀 🚀 🚀  Lets get the rust part done and merge this !

File: cli/flox-rust-sdk/src/models/environment/core_environment.rs [hunk 1]
```diff
@@ -579,9 +578,7 @@ pub enum CoreEnvironmentError {
     #[error("failed to update environment")]
     UpdateFailed(#[source] CallPkgDbError),
     // endregion
-
-    // might be merged with pkgdb soon
-    #[error("call to env-builder failed")]
+    #[error("call to `pkgdb buildenv' failed")]
```
Comment (ysndr) [inline cli/flox-rust-sdk/src/models/environment/core_environment.rs:581]:
```suggestion
    #[error("call to `pkgdb buildenv` failed")]
```

File: pkgs/flox-cli/default.nix [hunk 2]
```diff
@@ -28,7 +28,7 @@
   pkgsFor,
   pre-commit-check,
   flox-pkgdb,
-  flox-env-builder,
+  longVersion ? false,
```
Comment (garbas) [inline pkgs/flox-cli/default.nix:31]: `longVersion` can be removed, we're not using it anymore.

Comment (aakropotkin) [reply pkgs/flox-cli/default.nix]: I dropped it. Thanks

**Comment by mkenigs** [comment]
This is a big ol diff so I'll come back and review if @ysndr or @garbas aren't about to approve it

**Comment by aakropotkin** [comment]
> This is a big ol diff so I'll come back and review if @ysndr or @garbas aren't about to approve it

Yeah sorry about the size of the thing.

It's moving a lot of code, and had to execute changes to the build/infra at the same time to work.
It's not easily broken up.

**Review by mkenigs — CHANGES_REQUESTED** [review]
Overall changes look good. `just build` is failing for me (with the `fix: Priority ctx on darwin` commit)
```
/nix/store/pkbcb46dxzpkf4zhkl094cn08ql7jh4h-libcxx-11.1.0-dev/include/c++/v1/memory:1695:36: error: no matching constructor for initialization of 'flox::buildenv::RealisedPackage'
                ::new ((void*)__p) _Tp(_VSTD::forward<_Args>(__args)...);
                                   ^   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/pkbcb46dxzpkf4zhkl094cn08ql7jh4h-libcxx-11.1.0-dev/include/c++/v1/memory:1538:14: note: in instantiation of function template specialization 'std::__1::allocator_traits<std::__1::allocator<flox::buildenv::RealisedPackage>>::__construct<flox::buildenv::RealisedPackage, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, bool, flox::buildenv::Priority>' requested here
            {__construct(__has_construct<allocator_type, _Tp*, _Args...>(),
             ^
/nix/store/pkbcb46dxzpkf4zhkl094cn08ql7jh4h-libcxx-11.1.0-dev/include/c++/v1/vector:926:21: note: in instantiation of function template specialization 'std::__1::allocator_traits<std::__1::allocator<flox::buildenv::RealisedPackage>>::construct<flox::buildenv::RealisedPackage, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, bool, flox::buildenv::Priority>' requested here
    __alloc_traits::construct(this->__alloc(), _VSTD::__to_address(__tx.__pos_),
                    ^
/nix/store/pkbcb46dxzpkf4zhkl094cn08ql7jh4h-libcxx-11.1.0-dev/include/c++/v1/vector:1687:9: note: in instantiation of function template specialization 'std::__1::vector<flox::buildenv::RealisedPackage, std::__1::allocator<flox::buildenv::RealisedPackage>>::__construct_one_at_end<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, bool, flox::buildenv::Priority>' requested here
        __construct_one_at_end(_VSTD::forward<_Args>(__args)...);
        ^
src/buildenv/realise.cc:254:16: note: in instantiation of function template specialization 'std::__1::vector<flox::buildenv::RealisedPackage, std::__1::allocator<flox::buildenv::RealisedPackage>>::emplace_back<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, bool, flox::buildenv::Priority>' requested here
          pkgs.emplace_back(
               ^
/Users/matthew/flox/main/pkgdb/include/flox/buildenv/realise.hh:41:8: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 3 were provided
struct RealisedPackage
       ^
/Users/matthew/flox/main/pkgdb/include/flox/buildenv/realise.hh:41:8: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 3 were provided
/Users/matthew/flox/main/pkgdb/include/flox/buildenv/realise.hh:41:8: note: candidate constructor (the implicit default constructor) not viable: requires 0 arguments, but 3 were provided
1 error generated.
make: *** [<builtin>: src/buildenv/realise.o] Error 1
make: Leaving directory '/Users/matthew/flox/main/pkgdb'
error: Recipe `build-pkgdb` failed on line 27 with exit code 2
```

File: Justfile [hunk 3]
```diff
@@ -38,38 +36,39 @@ build: build-cli
...
-test-env-builder: build-env-builder
-    @pushd env-builder; make -j tests; popd
-    @pushd env-builder; make check; popd
+    @make -C pkgdb -j tests;
+    @make -C pkgdb check;
 
 # Run the end-to-end test suite
 functional-tests +bats_args="": build
-    @flox-tests {{bats_args}}
+    @flox-tests {{bats_args}};
```
Comment (mkenigs) [inline Justfile:44]: I think we should start using `shfmt`, but until then I'd prefer if we don't fight over `;` in diffs

**Review by mkenigs — APPROVED** [review]

**Pull request merged**